src/docs/patterns/user-auth.mdx

```diff
@@ -0,0 +1,296 @@
+
+<!-- @TODO switch login links over to the "Component Recommendations style" instead of an "Interactive Code example" once we have one designed. -->
+
+<React.Fragment>
+    <DemoCard repository={'login'} angular react ionic reactNative />
+    <Divider />
+</React.Fragment>
+
+# Variations
+
+<ImageGrid
+    images={[ViewPortLoginImage, ViewPortOverlayImage]}
+    regularWidth
+    caption={[
+        `User authentication workflows demand users' attention. Note that the workflows are usually "boxed" in a card.`,
+        `1&2: An example of a login workflow, on mobile and on desktop.`,
+        `3&4: An example of a password changing workflow, on mobile and on desktop`,
+        `Note that #2 represents a typical desktop auth layout when the user is NOT logged in, while #4's layout is typical for the workflow when the user has already logged in.`,
+    ]}
+/>
+
+Your user authentication workflow should work on different screen sizes. On mobile, it should take up an entire screen. On a desktop, it is usually "boxed" in a card. The content should follow the rules specified by our [forms and validation pattern](/patterns/forms), and each page should be connected using our [steppers pattern](/patterns/steppers).
+
+<MaterialDesignDescription 
+    title={'Forms and Validation'}
+    url={'/patterns/forms'}
+    description={`Follow PX Blue's guideline for form controls.`}
+     avatar={<PxblueSmallAltIcon style={{ width: 48, height: 48 }} />}
+/>
+<MaterialDesignDescription 
+    title={'Steppers'}
+    url={'/patterns/steppers'}
+    description={`Follow PX Blue's guideline for navigating between pages.`}
+     avatar={<PxblueSmallAltIcon style={{ width: 48, height: 48 }} />}
+/>
+
+Currently, PX Blue supports the following authentication workflows:
+
+-   Registration
+-   Join organization
+-   Login
+-   Reset password
+
+We will now discuss them one by one.
+
+## Registration
+
+Registration is the process of user account creation. Depends on your project setup, the user might need to request to join various organizations, which we will discuss later. Registration includes the following variations:
+
+**Self-Registration**
+
+Self-registration involves registering an account without an email invitation. The workflow involves the following screens:
+
+-   Enter email address
+-   Accept EULA
+-   Verify email
+-   Create password
+-   Add account details
+-   Success & welcome
+
+**Registration via Invitation**
+
+Sometimes the organization admin might email the user an invitation. The workflow involves the following screens:
+
+-   Click through an email invitation
+-   Accept EULA
+-   Create password
+-   Add account details
+-   Success & welcome
+
+### Screens
+
+Most steps involved in self-registration and registration via invitation are very similar. Use the following screens for the registration workflow where appropriate.
+
+<ImageGrid
```

Comment:
Should we show desktop version as well since they are a little different in terms of how the header looks?

Reply:
🤔  I personally think the images at the very beginning of the grids section is good enough.

Reply:
I would suggest mixing in a couple desktop ones to be more explicit and not force adopters to do too much extrapolation. The one at the beginning of the page (change password) is a single screen, so we don't currently have any examples of what a multi-screen workflow looks like on Desktop.

Reply:
We don't need a desktop version for every image, but it might be nice to have a few sprinkled in or alternate, etc.